lib/SILGen/SILGenTopLevel.cpp

```diff
@@ -214,7 +214,23 @@ void SILGenFunction::emitCallToMain(FuncDecl *mainFunc) {
     B.createReturn(loc, exitCode);
   } else {
     FuncDecl *exitFuncDecl = SGM.getExit();
-    assert(exitFuncDecl && "Failed to find exit function declaration");
+    if (!exitFuncDecl) {
```

Comment:
Do we have a test that hits this new code path? I'm wondering if `exitFuncDecl` is really valid to use when it hasn't been added to the module it's supposed to be in, or if there might be some other non-obvious problem with the generated AST.

Reply:
Yes, when I had it wrong there were some tests that were failing in Linux. Actually I think it just crashed the compiler during testing IIRC